clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaDownstreamMessage.java

```diff
@@ -69,6 +71,8 @@ public KafkaDownstreamMessage(final KafkaConsumerRecord<String, Buffer> record)
         payload = record.value();
         creationTime = getCreationTime(record.headers());
         timeTillDisconnect = getTimeTillDisconnect(record.headers());
+        correlationId = getCorrelationId(record.headers());
+        status = getStatus(record.headers());
```

Comment:
FMPOV we should not try to retrieve these values as part of the constructor because by far the most messages will not be command responses and we will simply be waisting CPU and generate useless log entries ...
Instead, we should retrieve the values when the getters are invoked or at least lazily initialize the fields.

Reply:
I have updated now, so that those values are being retrieved only when the _getters_ are invoked.